### PR TITLE
added connecting snackbar and snackbar.dart file

### PR DESCRIPTION
**Pull request by satyam372**
Added 'snackbar.dart' file to show the snackbar. I have created this file inside widget folder and then imported it in app.dart and message_list.dart file . 

working : I have used "connectivity_plus: ^5.0.2" package to implement this.   This package allows developers to see the connectivity of device . This helps us to determine  whether our device has active internet connection or not .

Video link:-
https://github.com/zulip/zulip-flutter/assets/112092492/911a49a4-568c-4b29-82bc-1c8d0bf5872b

### CODE REVIEW

File: lib/widgets/app.dart [hunk 1]
```diff
@@ -113,49 +115,64 @@ class ZulipApp extends StatelessWidget {
...
-          onGenerateInitialRoutes: (_) {
-            return [
-              MaterialWidgetRoute(page: const ChooseAccountPage()),
-              if (initialAccountId != null) ...[
-                HomePage.buildRoute(accountId: initialAccountId),
-                InboxPage.buildRoute(accountId: initialAccountId),
-              ],
-            ];
-          });
+        child: Builder(builder: (context) {
```
Comment (Khader-1) [inline lib/widgets/app.dart:118]: Update your code editors spacing to match the repo configuration. check the [ReadMe File](https://github.com/zulip/zulip-flutter?tab=readme-ov-file#code-formatting)

File: pubspec.yaml [hunk 2]
```diff
@@ -24,8 +24,8 @@ environment:
   # that by the time we want to release, these will have become stable.
   # TODO: Before general release, switch to stable Flutter and Dart versions,
   #   or pin exact versions: https://github.com/zulip/zulip-flutter/issues/15
-  sdk: '>=3.4.0-256.0.dev <4.0.0'
-  flutter: '>=3.21.0-12.0.pre.26'
+  sdk: '>=3.4.0-140.0.dev <4.0.0'
```
Comment (Khader-1) [inline pubspec.yaml:27]: Never change the sdk version in pubspec.yaml as it is configured for the whole team. refer to [ReadMe](https://github.com/zulip/zulip-flutter?tab=readme-ov-file#flutter-version)

File: pubspec.yaml [hunk 3]
```diff
@@ -38,6 +38,7 @@ dependencies:
     sdk: flutter
   flutter_localizations:
     sdk: flutter
+  fluttertoast: ^8.2.4
```
Comment (Khader-1) [inline pubspec.yaml:41]: As mentioned by @gnprice in [the stream](https://chat.zulip.org/#narrow/stream/48-mobile/topic/.23F465.20Show.20.22Connecting.E2.80.A6.22.20banner.20.28or.20equivalent.29.20when.20serv.2E.2E.2E/near/1763901) no need to use `fluttertoast`

**Review by Khader-1 — CHANGES_REQUESTED** [review]
You're doing good but you need to refer to the contribution guides linked in the comments.

**Comment by Khader-1** [comment]
Nice work but it can be improved, according to the [commits discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) your commits need some restructure.

**Comment by satyam372** [comment]
thankyou @Khader-1 for your feedback . appreciated your inputs and i will definetly work on it

**Comment by gnprice** [comment]
Thanks @Khader-1 for the initial review! And thanks @satyam372 for the contribution.

For cross-reference: it looks like this started as #583 and then became #586.

@satyam372 For future pull requests, it's better to update the existing PR than to close it and open a new one. See https://zulip.readthedocs.io/en/latest/git/pull-requests.html#update-a-pull-request .

**Pull request closed without merging**